src/components/scheduling/SchedulingGoalsPanel.svelte

```diff
@@ -80,22 +80,48 @@
 
   async function onUpdateGoal(event: CustomEvent<SchedulingGoalPlanSpecification>) {
     const {
-      detail: { goal_metadata, specification_id, ...goalPlanSpec },
+      detail: { goal_metadata, ...goalPlanSpec },
     } = event;
 
     if ($plan) {
       await effects.updateSchedulingGoalPlanSpecification(
         $plan,
-        specification_id,
         {
           ...goalPlanSpec,
+        },
+        user,
+      );
+    }
+  }
+
+  async function duplicateGoalInvocation(event: CustomEvent<SchedulingGoalPlanSpecification>) {
+    const {
+      detail: { goal_metadata, specification_id, ...goalPlanSpec },
+    } = event;
+
+    if ($plan) {
+      await effects.createSchedulingGoalPlanSpecification(
+        {
+          ...goalPlanSpec,
+          goal_invocation_id: undefined, // let DB autogenerate
```

Comment:
My only concern with setting `goal_invocation_id` to undefined here is that there's a chance that we may forget to set it to `undefined` in other places that use `effects.createSchedulingGoalPlanSpecification`. It might be more appropriate to explicitly make the `SchedulingGoalPlanSpecInsertInput` type not take a `goal_invocation_id`.

Reply:
Good point. I've made a new `SchedulingGoalPlanSpecInvocationInsertInput` that `Omit`'s `goal_invocation_id`, since `SchedulingGoalPlanSpecInsertInput` gets used by other effects that need to update specs, and would therefore need to specific a specific invocation id